### PR TITLE
avoid audit logs with unexpected errors

### LINKED ISSUE #20513: Audit log with unable to heal 0 corrupted blocks
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

I was trying to understand a high `minio_heal_objects_heal_total` metric and looking at the audit logs I have hundreds of thousands of lines `unable to heal 0 missing blocks on drives` and `unable to heal 0 corrupted blocks on drive`

I have the case on bucket objects but also on "systems" objects such as the `.minio.sys` bucket and `buckets/mypersonalbucket/.usage-cache.bin` objects

## Expected Behavior

I'm not sure it should log it couldn't heal 0 blocks, it means there's nothing to heal ?

## Current Behavior

N/A

## Possible Solution

N/A

## Steps to Reproduce (for bugs)

N/A

## Context

I'm trying to understand an elevated heal metrics

## Regression

I think it's a regression [introduced here](https://github.com/minio/minio/commit/e7a56f35b9bc8e782a4b0864c92627dfab2b3853) which is included in the version I'm running

## Your Environment

* Version used (`minio --version`): minio version RELEASE.2024-08-26T15-33-07Z (commit-id=1c4d28d7af0373cde810703839a179739dbe6f34)
* Server setup and configuration: 4x 24
* Operating System and version (`uname -a`): `Ubuntu 22.04` / `Linux minio1-node2.pr.par5.numberly.net 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
avoid audit logs with unexpected errors

## Motivation and Context
fixes #20513

## How to test this PR?
Looks like this is a regression from #20256 log.Error is incorrectly filled

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression from #20256
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by poornas — APPROVED** [review]

File: cmd/erasure-healing.go [hunk 1]
```diff
@@ -234,11 +234,12 @@ func (er *erasureObjects) auditHealObject(ctx context.Context, bucket, object, v
 	}
 
 	b, a := result.GetCorruptedCounts()
-	if b == a {
+	if b > 0 && a > 0 && b == a {
```
Comment (krisis) [inline cmd/erasure-healing.go]:
```suggestion
	if b > 0 && b == a {
```

File: cmd/erasure-healing.go [hunk 2]
```diff
@@ -234,11 +234,12 @@ func (er *erasureObjects) auditHealObject(ctx context.Context, bucket, object, v
...
 		opts.Error = fmt.Sprintf("unable to heal %d corrupted blocks on drives", b)
 	}
+
 	b, a = result.GetMissingCounts()
-	if b == a {
+	if b > 0 && a > 0 && b == a {
```
Comment (krisis) [inline cmd/erasure-healing.go]:
```suggestion
	if b > 0 && b == a {
```

**Review by krisis — APPROVED** [review]
LGTM

**Review by klauspost — APPROVED** [review]
👍🏼 with KPs simplification.

**Pull request merged**